ext/ffi/lib.rs

```diff
@@ -535,322 +569,808 @@ where
+          as i32;
+        ffi_args.push(NativeValue { i32_value: value });
+      }
+      NativeType::U64 => {
+        let value: u64 = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.u64_value().0
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI u64 type, expected number")
+          })? as u64
+        };
+        ffi_args.push(NativeValue { u64_value: value });
+      }
+      NativeType::I64 => {
+        let value: i64 = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.i64_value().0
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI i64 type, expected number")
+          })? as i64
+        };
+        ffi_args.push(NativeValue { i64_value: value });
+      }
+      NativeType::USize => {
+        let value: usize = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.u64_value().0 as usize
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI usize type, expected number")
+          })? as usize
+        };
+        ffi_args.push(NativeValue { usize_value: value });
+      }
+      NativeType::ISize => {
+        let value: isize = if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value)?;
+          value.i64_value().0 as isize
+        } else {
+          value.integer_value(scope).ok_or_else(|| {
+            type_error("Invalid FFI isize type, expected number")
+          })? as isize
+        };
+        ffi_args.push(NativeValue { isize_value: value });
+      }
+      NativeType::F32 => {
+        let value = value
+          .number_value(scope)
+          .ok_or_else(|| type_error("Invalid FFI f32 type, expected number"))?
+          as f32;
+        ffi_args.push(NativeValue { f32_value: value });
+      }
+      NativeType::F64 => {
+        let value = value
+          .number_value(scope)
+          .ok_or_else(|| type_error("Invalid FFI f64 type, expected number"))?
+          as f64;
+        ffi_args.push(NativeValue { f64_value: value });
+      }
+      NativeType::Pointer => {
+        if value.is_null() {
+          let value: *const u8 = ptr::null();
+          ffi_args.push(NativeValue { pointer: value })
+        } else if value.is_big_int() {
+          let value = v8::Local::<v8::BigInt>::try_from(value).unwrap();
+          let value = value.u64_value().0 as *const u8;
+          ffi_args.push(NativeValue { pointer: value });
+        } else if value.is_array_buffer() || value.is_array_buffer_view() {
+          let value: ZeroCopyBuf = serde_v8::from_v8(scope, value)?;
+          let value: &[u8] = &value[..];
+          ffi_args.push(NativeValue {
+            pointer: value.as_ptr(),
+          });
```

Comment:
UB: `value` is dropped at the end of this function, what you push to `ffi_args` is a dangling pointer

Reply:
ZeroCopyBuf, as its name suggests, does not copy the data from the `ArrayBuffer` but instead acts as a wrapper that can be used to reference into the memory. ZeroCopyBuf is dropped, but the `ArrayBuffer` lives until V8 decides it's time for it to be GC'd.

So, this definitely can be UB if the user passes in an AB that is then immediately dropped and a GC triggers. Otherwise it's totally fine.